src/components/structures/TimelinePanel.js

```diff
@@ -1023,7 +1020,16 @@ var TimelinePanel = React.createClass({
             var boundingRect = node.getBoundingClientRect();
             if ((allowPartial && boundingRect.top < wrapperRect.bottom) ||
                 (!allowPartial && boundingRect.bottom < wrapperRect.bottom)) {
-                return i;
+                // find last invisible event before the next visible one.
```

Comment:
Actually this should surely be done in `messagePanel.getNodeForEventId`, by returning falsey if the event is hidden